src/bridges/angle/AngleBridge.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Aztec.
+pragma solidity >=0.8.4;
+
+import {SafeERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {AztecTypes} from "../../aztec/libraries/AztecTypes.sol";
+import {ErrorLib} from "../base/ErrorLib.sol";
+import {BridgeBase} from "../base/BridgeBase.sol";
+import {IStableMaster} from "../../interfaces/angle/IStableMaster.sol";
+
+/**
+ * @title Angle Protocol bridge contract.
+ * @author Angle Protocol Team
+ * @notice Allows a user to deposit/withdraw his tokens to/from Angle Protocol
+ * @dev Enter Angle Protocol as an SLP by depositing authorized tokens
+ */
+contract AngleBridge is BridgeBase {
+    using SafeERC20 for IERC20;
+
+    IStableMaster public constant STABLE_MASTER = IStableMaster(0x5adDc89785D75C86aB939E9e15bfBBb7Fc086A87);
+
+    // The amount of dust to leave in the contract
+    // Optimization based on EIP-1087
+    uint256 internal constant DUST = 1;
+
+    /**
+     * @notice Set address of rollup processor
+     * @param _rollupProcessor Address of rollup processor
+     */
+    constructor(address _rollupProcessor) BridgeBase(_rollupProcessor) {
+        _preApprove(0xc9daabC677F3d1301006e723bD21C60be57a5915);
+        _preApprove(0xe9f183FC656656f1F17af1F2b0dF79b8fF9ad8eD);
+        _preApprove(0x3f66867b4b6eCeBA0dBb6776be15619F73BC30A2);
+        _preApprove(0x6b4eE7352406707003bC6f6b96595FD35925af48);
+    }
+
+    /**
+     * @notice A function which returns an _totalInputValue amount of _inputAssetA
+     * @param _inputAssetA - ERC20/ETH (deposit), or sanToken (withdraw)
+     * @param _outputAssetA - sanToken (deposit), or ERC20/ETH (withdraw)
+     * @param _totalInputValue - amount of ERC20/ETH to deposit, or the amount of sanToken to withdraw
+     * @param _auxData - 0 (deposit), 1 (withdraw)
+     * @return outputValueA - the amount of output asset to return
+     */
+    function convert(
+        AztecTypes.AztecAsset memory _inputAssetA,
+        AztecTypes.AztecAsset memory,
+        AztecTypes.AztecAsset memory _outputAssetA,
+        AztecTypes.AztecAsset memory,
+        uint256 _totalInputValue,
+        uint256,
+        uint64 _auxData,
+        address
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 outputValueA,
+            uint256,
+            bool
+        )
+    {
+        if (_inputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) revert ErrorLib.InvalidInputA();
+        if (_outputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) revert ErrorLib.InvalidOutputA();
+        if (_totalInputValue < 10) revert ErrorLib.InvalidInputAmount();
+
+        uint256 totalInputValue = _totalInputValue;
+
+        if (_auxData == 0) {
+            address poolManager = getPoolManager(_inputAssetA.erc20Address);
```

Comment:
It looks "almost" similar but it's not. Every line is different, going back and forth between input and output asset. So i think it's better to leave it like this, otherwise I would have an `if` on every line (or a ternary) and it will look confusing